### PR TITLE
SLCIE-116 - NPE thrown if path or resource is null

### PR DESCRIPTION
**Pull request by kwatral**

### CODE REVIEW

File: slice-test/slice-core-test/src/test/groovy/com/cognifide/slice/core/provider/SliceModelProviderTest.groovy [hunk 1]
```diff
@@ -210,38 +219,42 @@ class SliceModelProviderTest extends BaseSetup {
 	}
 
 	def "Get model by a null path"() {
-		def nullPath = null
+		String nullPath = null
+		setup: "Creating model for null path"
+		SimpleModel model = modelProvider.get(SimpleModel.class, nullPath);
```
Comment (mmajchrzak) [inline slice-test/slice-core-test/src/test/groovy/com/cognifide/slice/core/provider/SliceModelProviderTest.groovy]: Why have you removed the when-then structure?

File: slice-test/slice-core-test/src/test/groovy/com/cognifide/slice/core/provider/SliceModelProviderTest.groovy [hunk 2]
```diff
@@ -210,38 +219,42 @@ class SliceModelProviderTest extends BaseSetup {
...
-		thrown(ProvisionException)
+		and: "Model should have a null value for property 'prop1' set"
+		Assert.assertNull(model.getProp1())
 	}
 
 	def "Get model resource by key and resource - non-existing (null) resource"() {
 		def nonExistingPath = "/content/nonexisting"
 
-		setup: "Get non-existing resource"
+		setup: "Creating model for path: " + nonExistingPath
```
Comment (mmajchrzak) [inline slice-test/slice-core-test/src/test/groovy/com/cognifide/slice/core/provider/SliceModelProviderTest.groovy]: Seeing "Creating model for path: /content/nonexisting" in test output is not that informative as "Get non-existing resource". Use when-then as it was previously.

File: slice-test/slice-core-test/src/test/groovy/com/cognifide/slice/core/provider/SliceModelProviderTest.groovy [hunk 3]
```diff
@@ -153,10 +152,20 @@ class SliceModelProviderTest extends BaseSetup {
...
 
-		when: "Creating model for paths: " << nonExistingPath1 << nonExistingPath2
-		modelProvider.getListFromResources(SimpleModel.class, nonExistingResources.iterator())
-		then: "Throw ProvisionException - either path or resource should be set"
-		thrown(ProvisionException)  //FIXME we should not expect ProvisionException here
+		setup: "Creating model for paths: " << nonExistingPath1 << nonExistingPath2
+		List<SimpleModel> models = modelProvider.getListFromResources(SimpleModel.class, nonExistingResources.iterator())
+
+		expect: "List of models should not be null"
+		Assert.assertNotNull(models)
```
Comment (mleczey) [inline slice-test/slice-core-test/src/test/groovy/com/cognifide/slice/core/provider/SliceModelProviderTest.groovy]: Could you make static import for ```Assert```?

Comment (michalwronowski) [inline slice-test/slice-core-test/src/test/groovy/com/cognifide/slice/core/provider/SliceModelProviderTest.groovy]: No, this way it is consistent with other tests.

**Pull request merged**